src/Session.js

```diff
@@ -562,27 +548,27 @@ Session.prototype = {
   receiveReinvite: function(request) {
     var self = this;
 
-    if (!request.body) {
-      return;
-    }
-
-    if (request.getHeader('Content-Type') !== 'application/sdp') {
+    if (!this.mediaHandler.hasDescription(request)) {
       this.logger.warn('invalid Content-Type');
       request.reply(415);
       return;
     }
 
-    this.mediaHandler.setDescription(request.body)
+    this.mediaHandler.setDescription(request)
     .then(this.mediaHandler.getDescription.bind(this.mediaHandler, this.mediaHint))
-    .then(function(body) {
-      request.reply(200, null, ['Contact: ' + self.contact], body,
+    .then(function(description) {
+      var extraHeaders = ['Contact: ' + self.contact];
+      if (description.extraHeaders) {
+        extraHeaders = extraHeaders.concat(description.extraHeaders);
+      }
+      request.reply(200, null, extraHeaders, description.body,
```

Comment:
Yes, the idea is that `getDescription` returns enough information to be able to set a Content-Type header (and potentially other headers) using the `extraHeaders` property, and to set the body using the `body` property. I am not sure if this should be called `description`, but it is all I could think of at the time.

Another approach I thought of was to return a function that transforms the outgoing SIP message, but I don't think SIP.js is as well-suited to that approach.